src/Admin/AbstractAdmin.php

```diff
@@ -3531,6 +3515,38 @@ protected function configureDefaultSortValues(array &$sortValues)
     {
     }
 
+    /**
+     * Set the parent object, if any, to the provided object.
+     */
+    final protected function appendParentObject(object $object): void
+    {
+        if ($this->isChild() && $this->getParentAssociationMapping()) {
+            $parentAdmin = $this->getParent();
+            $parentObject = $parentAdmin->getObject($this->request->get($parentAdmin->getIdParameter()));
+
+            if (null !== $parentObject) {
+                $propertyAccessor = $this->getConfigurationPool()->getPropertyAccessor();
+                $propertyPath = new PropertyPath($this->getParentAssociationMapping());
+
+                $value = $propertyAccessor->getValue($object, $propertyPath);
+
+                if (\is_array($value) || $value instanceof \ArrayAccess) {
+                    $value[] = $parentObject;
+                    $propertyAccessor->setValue($object, $propertyPath, $value);
+                } else {
+                    $propertyAccessor->setValue($object, $propertyPath, $parentObject);
+                }
+            }
+        } elseif ($this->hasParentFieldDescription()) {
+            $parentAdmin = $this->getParentFieldDescription()->getAdmin();
+            $parentObject = $parentAdmin->getObject($this->request->get($parentAdmin->getIdParameter()));
```

Comment:
Better open an issue, this PR is 2 years old.

But, we won't check embedded in SonataAdmin since it's not related to any persistence bundle.

getObject is doing
```
final public function getObject($id): ?object
    {
        if (null === $id) {
            return null;
        }

        $object = $this->getModelManager()->find($this->getClass(), $id);
        if (null === $object) {
            return null;
        }

        $this->alterObject($object);
        foreach ($this->getExtensions() as $extension) {
            $extension->alterObject($this, $object);
        }

        return $object;
    }
```
Where does come from the Exception ?

If it's in the 
```
$this->getModelManager()->find($this->getClass(), $id);
```
call, we could add some check in https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/4.x/src/Model/ModelManager.php

But its weird to have an entity without identifier, since all the admin (and parent admin) are supposed to be related to an entity/document.